### DISCUSSION TITLE
Props has type "any"

### DISCUSSION THREAD

**Discussion by ThomasBnf** [next-safe-action/next-safe-action#196] [Q&A] [answered]
Hi,

Yesterday, when I restarted coding, all my props had the type "any." It was impossible to know what happened. I didn't install any new packages. I tried rolling back to my last commit, and after VS Code loaded, all the type errors appeared...

I am building my server action with V6.

This is my SafeActionClient

```js
import { createSafeActionClient } from "next-safe-action";

export const safeAction = createSafeActionClient();
```

This is my SafeAuthActionClient

```js
import { auth } from "@/auth";
import { UserSchema } from "@/schema/UserSchema";
import { createSafeActionClient } from "next-safe-action";
import { prisma } from "./prisma";

export const safeAuthAction = createSafeActionClient({
  async middleware() {
    const session = await auth();

    if (!session?.user?.id) {
      throw new Error("Session not found!");
    }

    const user = UserSchema.parse(
      await prisma.user.findUnique({
        where: { id: session.user.id },
        include: {
          role: true,
        },
      }),
    );

    return {
      userId: user.id,
      companyId: user?.currentCompany,
      currentUser: user,
    };
  },
});
```

this is an example of server action 

```js
"use server";

import { prisma } from "@/lib/prisma";
import { safeAuthAction } from "@/lib/safeAuthAction";
import { InterventionSchema } from "@/schema/InterventionSchema";
import { ValueSchema } from "@/schema/ValueSchema";
import { revalidatePath } from "next/cache";
import { z } from "zod";

export const createIntervention = safeAuthAction(
  z.object({
    values: z.array(ValueSchema),
    path: z.string(),
  }),
  async ({ values, path }, { userId, companyId }) => {
    const intervention = await prisma.intervention.create({
      data: {
        userId,
        companyId,
        values,
      },
    });

    revalidatePath(path);
    return InterventionSchema.parse(intervention);
  },
);
```

In this query, values and path as "any" as types...

**Comment by TheEdoRan** [comment]
You're probably using TypeScript 5.5. Unfortunately TypeSchema is not compatible with it, so rollback to 5.4.

**Reply by ThomasBnf** [reply]
This is my package.json dev devDependencies.

what does "^5" means ? I think i'm on typescript 5 no ?

 "devDependencies": {
    "@types/bcryptjs": "^2.4.6",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-datepicker": "^6.2.0",
    "@types/react-dom": "^18",
    "@types/react-signature-canvas": "^1.0.5",
    "eslint": "^8",
    "eslint-config-next": "14.2.3",
    "partykit": "0.0.105",
    "postcss": "^8",
    "prettier": "^3.2.5",
    "prettier-plugin-tailwindcss": "^0.5.14",
    "prisma": "^5.14.0",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }

**Reply by ThomasBnf** [reply]
@TheEdoRan I tried to install 5.4 but i still have this issue ...

**Reply by TheEdoRan** [reply, marked as answer]
Your VS Code is probably using the default editor TS version. Use the workspace one and then reload the window, the error should go away.

**Reply by ThomasBnf** [reply]
@TheEdoRan Thanks ! it works

**Reply by hoaht-8203** [reply]
I face it too when update the vscode! Thanks

> Your VS Code is probably using the default editor TS version. Use the workspace one and then reload the window, the error should go away.
